### PR TITLE
RM changes for PR#2603

### PR DESCRIPTION
**Pull request by davidcok**
<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/DafnyRef/UserGuide.md [hunk 1]
```diff
@@ -101,36 +101,63 @@ on the [Dafny INSTALL page in the wiki](https://github.com/dafny-lang/dafny/wiki
...
+Each command may have its own subcommands and its own options, in addition to generally applicable options. 
+Thus the format of the command-line is
+a command name, followed by 0 or more subcommands, followed by options and files:
+`dafny <command> <subcommand>* <options> <files>`
+
+The command-line `dafny <command> --help` gives help information for that particular <command>.
+
+Also, the command-style command-line has modernized the syntax of options. Like many other tools, options
+now typically begin with a double hyphen, with some options having a single-hyphen short form, such as `--help` and `-h`.
+The value of an option is given after an `=` character, rather than a `:`, as in `--target=java`
```
Comment (keyboardDrummer) [inline docs/DafnyRef/UserGuide.md:118]: You can use a space, `:` or `=` as the delimiter between an option and its argument.

File: docs/DafnyRef/UserGuide.md [hunk 2]
```diff
@@ -101,36 +101,63 @@ on the [Dafny INSTALL page in the wiki](https://github.com/dafny-lang/dafny/wiki
...
+
+If no command is given, then the command-line is presumed to use old-style syntax, so any previously 
+written command-line will still be valid.
+
+The following  commands are recognized. In each case all files listed are parsed and typechecked.
```
Comment (keyboardDrummer) [inline docs/DafnyRef/UserGuide.md:123]: double space

File: docs/DafnyRef/UserGuide.md [hunk 3]
```diff
@@ -101,36 +101,63 @@ on the [Dafny INSTALL page in the wiki](https://github.com/dafny-lang/dafny/wiki
...
+This is similar to the old-style options `/compile:3` or `/compile:4`.
+- `dafny translate` -- verifies the program and translates it to a source artifact, perhaps with information for
+a target language build tool, for the target platform, similar to the previous `/compile:0 '/spillTargetCode:2`.
+`--no-verify` suppresses verification checks.
+- `dafny build` -- verifies (except if `--no-verify` is used) and compiles the Dafny program, producing an executable artifact for the target system. The command has options that enable being specific about the build platform and architecture.
+
+The command-line also expects the following:
 - Files are designated by absolute paths or paths relative to the current
-working directory. A command-line argument not matching a known option is considered a filepath.
+working directory. A command-line argument not matching a known option is considered a filepathi, and likely one
```
Comment (keyboardDrummer) [inline docs/DafnyRef/UserGuide.md:136]: extra `i`

File: docs/DafnyRef/UserGuide.md [hunk 4]
```diff
@@ -101,36 +101,63 @@ on the [Dafny INSTALL page in the wiki](https://github.com/dafny-lang/dafny/wiki
...
-default, but can be skipped by using the `-noVerify` or `-dafnyVerify:0` option
+default, but can be skipped by using the `-noVerify` or `--no-verify` option
 - Compiling the dafny program to a target language. This step is performed by default if the verification is
-successful but can be skipped or always executed by using variations of the `-compile` option.
-- Whether the source code of the compiled target is written out is controlled by `-spillTargetCode`
-- The particular target language used is chosen by `-compileTarget`
-- Whether or not the `dafny` tool attempts to run the compiled code is controlled by `-compile`
+successful but can be skipped or always executed by using variations of the `-compile` option or different commands.
+- Whether the source code of the compiled target is written out is controlled by `-spillTargetCode` or the top-level command that is being executed.
+- The particular target language used is chosen by `-compileTarget` or `--target`
```
Comment (keyboardDrummer) [inline docs/DafnyRef/UserGuide.md:160]: This depends on the mode. Do you think it's good to have help both for new and old mode? I imagine we would switch over help to the new mode at some point and then remove help for the old mode.

File: docs/DafnyRef/UserGuide.md (hunk 2 above)
Comment (davidcok) [reply docs/DafnyRef/UserGuide.md:123]: Fixed.

File: docs/DafnyRef/UserGuide.md (hunk 3 above)
Comment (davidcok) [reply docs/DafnyRef/UserGuide.md:136]: Fixed.

File: docs/DafnyRef/UserGuide.md (hunk 4 above)
Comment (davidcok) [reply docs/DafnyRef/UserGuide.md:160]: Agreed, but at the moment we need both -- until we deprecate the old style completely.

**Review by keyboardDrummer — APPROVED** [review]

**Pull request merged**